lib/Util.ts

```diff
@@ -75,4 +76,46 @@ function buildParamsStringifyOptions(queryParams: IRequestQueryParams): any  {
     }
 
     return options;
-}
\ No newline at end of file
+}
+
+/**
+ * Decompress/Decode gzip encoded JSON
+ * Using Node.js built-in zlib module
+ *
+ * @param {Buffer} buffer
+ * @param {string} charset? - optional; defaults to 'utf-8'
+ * @return {Promise<string>}
+ */
+export async function decompressGzippedContent(buffer: Buffer, charset?: string): Promise<string> {
+    return new Promise<string>(async (resolve, reject) => {
+        zlib.gunzip(buffer, function (error, buffer) {
+            if (error) {
+                reject(error);
+            }
+
+            resolve(buffer.toString(charset || 'utf-8'));
```

Comment:
Tested with a jsdelivr endpoint, zlib errors with `incorrect header check` and buffer here is undefined.

Reply:
I've just tested one of **jsDelivr**'s endpoints, specifically: `https://raw.githubusercontent.com/jsdelivr/jsdelivr/master/Privacy%20Policy%20-%20jsdelivr.net.md` which has GZipped (gzip encoded) content, and the string body was parsed neatly as string, with no Errors. Moreover, this PR included tests against GZipped contents endpoints, and they passed successfully. If you can, please provide the endpoint you've tried and provided the error above.